### DISCUSSION TITLE
Скрипты в фоне

### DISCUSSION THREAD

**Discussion by tipodanet** [k-ivan/visual-bookmarks-chrome#188] [Ideas]
Иван, вы как-то говорили  , что в фоне можно запустить скрипты , например от THREE. 
Вот тогда у вас будет юзеров сразу + 1000000.  Надо признать, большинству важен дизайн, а не поиски, селекты, менюшки.
Я где-то видел готовый скрипт из этого Приложения  MS STORE
[Lively Wallpaper](https://apps.microsoft.com/detail/9ntm2qc6qws7?hl=ru-ru&gl=RU)

![apps 21640 14416131676512756 8f975657-3f20-4265-8782-90e4e981cac8](https://github.com/user-attachments/assets/b80b8687-64ee-46ce-a238-45ea7d884856)
это самое навороченный скрипт для фона, что я видел . Вроде он бесплатный и открытый .

**Comment by tipodanet** [comment]
Да.  все просто. Ставим это  расширение.
Лично у меня на Windows Beta 26120.xxxx оно  запускается. Дергается, скрипт мигает  (но год назад на Win11-23H2  он работал). Значит , среда загрузилась, но Windows ее сразу убивает. 
Мои скромные выводы: Само расширение создает мини-Chromium  , типа браузер. И запускает в нем все, что хочет, в стандартных HTML, CCS, JS. Но, видно, этот мини-Chromium  , типа браузер недоделан. Или сделан под старую Windows. А новая  Windows 11 24H2 уже помощнее, и старье не показывает :)   По некоторым признакам, скрип(т)ы из 2020 года. 

Короче,  этот мини-Chromium  , типа браузер умирает сразу на запуске. 
Лезем в его исходники, что мы видим -- полный готовый перечень всех возможностей этого приложения [Lively Wallpaper](https://apps.microsoft.com/detail/9ntm2qc6qws7?hl=ru-ru&gl=RU). Там лежит все готовое. 
Этот скрипт я уже могу выложить себе на сайты. Для красоты. 
Там есть лицензии, ссылки на оригиналы. Этот скрипт  --красивая переделка другого монохромного скрипта.
В лицензиях видно, что автор  [Lively Wallpaper](https://apps.microsoft.com/detail/9ntm2qc6qws7?hl=ru-ru&gl=RU) не первый, и даже не второй. И все лежит под Общими лицензиями. 

Я уверен, что примерно 99% юзеров из домохозяек такую красоту  не видели (они вообще не видели слово "script"). Оставшийся 1% -- программеры. Из них 1% -- фронтендеры, лазящие по сайтам со скриптами. Всем остальным этот WebGL  не интересен.
Т.е.  Иван, можете подключать скрипты любые, хоть завтра :).

**Comment by k-ivan** [comment]
Если я решу добавить WebGL эффекты там точно не будет миллиона настроек и миллиона вариантов, максимум несколько параллакс или псевдо 3D для картинки на фоне(этого вполне достаточно), на этом все и точно не завтра. Это расширение для работы с закладками а не для любования миллионами анимаций. Если хочется необычного, найдите зацикленное видео с крутой анимацией и вставьте его фоном, оно и ресусры в большом кол-ве есть не будет и внешне красиво будет. Тащить что-то огромное я точно не буду.

**Comment by tipodanet** [comment]
у этого конкретного скрипта нет настроек ( или откл по умолчанию, инструкцию не читал , и здесь они не нужны)
Этот скрипт 106 Кб. На моем старом компе он есть 7% процессора. Видео 4К есть 20-40% процессора.
Есть скрипты ,  это конечно, тормозят, как 8К видео. Но не в этом случае. 
Большинство скриптов не едят процессор. 
И если выбудете постоянно вспоминать  дедушек , купивших в 00-х свой первый и единственный Celeron , то надо ориентироваться не на широкую аудиторию, а на дома престарелых.
Действительно, смешно иногда читать на Хабре комменты профи, вспоминающие IE11 , и  даже IE6 еще есть :) :) . Всегда заминусованные. Вы не на работе, тупого шефа рядом нет . 

Что значит " миллионами анимаций" .  Вы переворачиваете мои слова, так что  я не могу понять. 
Открытого поля для JS + CSS не будет , я уже понял, Google защищает  расширения браузера, но не защищает сам браузер. Через расширения скрипт запустить нельзя, а просто перетащить его в браузер  -- можно. Оставим это на совести Google.

Какие миллионы ? 
Скрипт один , количество 1 штука.
Автор [Lively Wallpaper](https://apps.microsoft.com/detail/9ntm2qc6qws7?hl=ru-ru&gl=RU) взял готовый чужой скрипт , взял еще 8 штук готовых чужих скриптов. 
И теперь он на 1 месте в MS STORE  в разделе "Wallpaper".

Иван, даже здесь, на Github , юзер где-то написал, что дизайн красивый , еще чего спрашивал по мелочи .

Вы немного витаете в IT-облаках, переоценивает людей. Думаете, что люди ищут прямо очень такие необходимые  им штуки из backend, какой-то сложный функционал .
Даже я полные настройки прочитал только через пару месяцев. 

Первый взгляд, работает, мои закладки на месте, лого есть, дизайн на месте  -- все ! Дальше мало кому интересно. 
Вы будете делать боковое меню полгода по вечерам.

Да, я его не видел нигде. Этим можно выделиться (Есть в Tabsbook, но там другой принцип)
Почему нет ? 2 варианта: 
1 -- сложно сделать
2 -- те авторы просто решили, что оно никому не надо. 
Это тоже вероятность реальная. 

Вы автор Визуальных Закладок, но Визуал ставите на последнее место. 
Пересмотрите приоритеты. Не только свои, а предположите, что есть и другие люди, с другими интересами.

**Reply by k-ivan** [reply]
Давайте уже закончим эти споры, я не считаю приоритетом данную тему. Давайте закроем ее, расширение будет обрастать функционалом работой с закладками, а не анимациями.

**Comment by tipodanet** [comment]
`параллакс или псевдо 3D`

далеко ходить не надо, :)  -- в этом расширения уже есть готовый и `параллакс,  и псевдо 3D` 
Даже я могу его взять и переделать для себя. Картинки заменить. Мне лицензии не нужны.

**Reply by k-ivan** [reply]
В таком случае вперед!

**Reply by tipodanet** [reply]
> В таком случае вперед!

Иван, что за детские обиды, 
Я не пойму. Вы как будто врага нашли, или я тут войну веду против Вас и ВЗ. 
Что общего  между моими тренировочными сайтами и Визуальным Закладками ? Что за обиды ? Я скрипт на дно ВЗ не положу.

Сайтов со скриптами сотни, половина из них с туториалами. Многие уже закрылись из-за низкой активности, 0 комментов. 
Эти скрипты ходят по сайтам-туториалам тыщами.
Открывается canvas  , script  --- готово.

> Давайте уже закончим эти споры, я не считаю приоритетом данную тему. Давайте закроем ее, расширение будет обрастать функционалом работой с закладками, а не анимациями.

Прекрасно. 

Вопрос: Каким ????

Я тут уже всю работу за вас выполнил по поддержке связей с общественностью. Что-то повышенной активности не наблюдается.
Причины ? 
1 -- Лень куда-то писать. Я кроме себя тут никого не наблюдаю, кто больше 2 мессаг настрочил.
2 -- люди пришли с работы. Виз Закладки крутятся, ближайшие 10 сайтов открываются. Все , Людям достаточно. Вывод: их все устраивает , Доп. функционал не требуется. 

Из 20000 здесь только я и пару людей по переводу. 

Я бы предложил создать Опрос "Какой вам , юзеры, нужен функционал ? "
Вот только, исходя их вышесказанного, не знаю, какие будет результаты опроса.

Если вы выбрали направление , так сообщите благодарным читателям, какого функционала нам ждать

**Reply by k-ivan** [reply]
Вы сами что-то придумывайте, нет никаких обид, я несколько раз пытался донести, что данную тему не считаю актуальной для закладок, поэтому проще закрыть.

**Comment by k-ivan** [comment]
Закрываю дискуссию, конструктива в ней нет.

**Reply by tipodanet** [reply]
Конструктива здесь выше крыши. И все за моим авторством. 
Тем более, эта тема Ваша. Это  Ваши слова "Скрипты на дно".
Тем более, тема скриптов и не предполагает никакого конструктива. 
Весь конструктив -- это имена скриптов, и ответ : берем / не берем.

Правда , я помню вы сказали ,что будет сами кодить :) :) 
Не понятно, зачем. Открытых и свободных от лицензий хватает и так всем

**Reply by k-ivan** [reply]
Не было у меня слов "Скрипты на дно", даже не понимаю, что означает эта фраза.

Давайте отвечу, касаемо этой темы - не берём.

**Reply by tipodanet** [reply]
"на дно" -- не было, "скрипты" -- были. А их некуда класть, кроме canvas. Без вариантов. А canvas где находится  ? :) 
Хорошее предложение Feance накатал.  [Customized Thumbnails ](https://github.com/k-ivan/visual-bookmarks-chrome/issues/93)
И даже это уже было. Было расширение FVD Speed Dail (вроде и есть), У него именно так и было, но чуть проще. Оно просто искало все .ico  и .png что в <meta     было. 

Именно такой функционал года 3 назад просили юзеры у Вивальди. Но у них просить, это  как со стеной ...

**Comment by tipodanet** [comment]
Как я уже сказал, мне эти скрипты не нужны. Я могу их за час себе скопировать.

А вот то, что благодарные домохозяйки, их не увидят . Это проблема , Иван, ваша ! И вашего отношения к своим ВЗ.
Вот мне реально интересен результат опроса: 
"Что выбрать для будущих ВЗ: 
1 -- Customized Thumbnails
2 -- Скрипты под Dials.

Я даже сразу уверен в его результатах --- общее мнение будет противоположно вашему. 

Создайте опрос, соберите ссылки всех тем (тут уже достаточно открытых и закрытых предложений)
Он будет висеть долго, но может через пару месяцев хоть 10 юзеров проголосует.
